src/controllers/AuthLocalController.js

```diff
@@ -10,65 +12,155 @@ dotenv.config();
+      message: 'user table is empty',
+      users
     });
-    if (!newUser.errors && newUser && Object.keys(newUser).length > 0) {
+  }
+
+  /**
+   * @description function for admin to create user
+   * @param {object} req user request object
+   * @param {object} res response object from server
+   * @returns {object} return true if user created or flase when was not
+   */
+  static async create(req, res) {
+    const { password } = req.body;
+    req.body.password = helper.password.hash(req.body.password);
+    const newUser = await User.create(req.body);
+    if (newUser.errors) {
+      const errors = helper.checkCreateUpdateUserErrors(newUser.errors);
+      const { code } = errors;
+      delete errors.code;
+      return res.status(code).json(errors);
+    }
+    if (newUser) {
       delete newUser.password;
+      await mailer(req.body.email, 'reset', req.body.firstName, password);
       return res.status(status.CREATED).send({
-        user: newUser,
-        token: helper.token.generate({ email, role: req.body.isAdmin })
+        message: `activetion message sent to ${req.body.email}`
       });
     }
   }
 
   /**
-   * @description - login user function
+   * @description function to activate user account
    * @param {object} req
    * @param {object} res
-   * @returns {object} user token
+   * @returns {object} it return true if account activeted otherwise it return false
    */
-  static async login(req, res) {
-    try {
-      const { email, password } = req.body;
-      const checkUser = await User.findOne({ email });
-      if (Object.keys(checkUser).length > 0) {
-        const comparePassword = helper.password.compare(password, checkUser.password);
-        if (!comparePassword) {
-          return res.status(status.UNAUTHORIZED).send({
-            message: 'The credentials you provided is incorrect'
-          });
-        }
-        const payload = { id: checkUser.id, role: checkUser.role };
-        const token = helper.token.generate(payload);
-        delete checkUser.password;
-        return res.status(status.OK).send({
-          user: checkUser,
-          token
-        });
-      }
-      return res
-        .status(status.UNAUTHORIZED)
-        .send({ error: `User with ${email} email doesn't exist!!!` });
-    } catch (error) {
-      return res.status(500).send({ error });
-    }
+  static async activate(req, res) {
+    const { user } = req;
+    const activated = await User.update({ isActive: true }, { email: user.email });
+    await mailer(activated.email, 'activated', activated.firstName, null);
+    return res.status(status.OK).send({
+      message: 'Account successfully activated'
+    });
+  }
+
+  /**
+   * @description function to activate user account
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} it return true if account verified otherwise it return false
+   */
+  static async reset(req, res) {
+    return res.status(status.OK).send({
```

Comment:
i thought the reset link is sent via email

Reply:
yes the reset link sent via email, I added this controller to return the message when user click on link

Reply:
resolved

Reply:
resolved